src/pages/MainPage/components/AddTodo/useAddTodoToDb.ts

```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+import { collection, addDoc } from "firebase/firestore";
+import db from "../../../../backend/firebase";
+import { checkTodo } from "./AddTodo.utils";
+
+const useAddTodoToDb = (todoDetails: ITodoDetails) => {
+  const [loading, setLoading] = useState(false);
+  const [status, setStatus] = useState<string[] | string>();
+
+  const submitTodo = async () => {
+    if (checkTodo(todoDetails).length > 0) {
+      setStatus(checkTodo(todoDetails));
+    } else {
+      setLoading(true);
+      try {
+        await addDoc(collection(db, "/todos"), {
+          task: todoDetails.task,
+          description: todoDetails.description || "",
+          name: todoDetails.name,
+          deadline: new Date(todoDetails.deadline),
+          createdAt: new Date(),
+          completed: false,
+          urgent: false,
+        });
+        setStatus("Success");
```

Comment:
albo, jeśli chcesz w zmiennej 'status' trzymać 'Success' (i być może 'failure' też) to warto dodać enuma i go trzymać w useState